dev-docs/RFCs/v7.x/heatmap-layer-rfc.md

```diff
@@ -0,0 +1,63 @@
+# RFC: Heatmap Layer
+
+* **Authors**: Ravi Akkenapally
+* **Date**: June 2019
+* **Status**: **Draft**
+
+
+## Abstract
+
+`deck.gl` provides several aggregation layers, `GridLayer`, `HexagonLayer` and `ContourLayers`. All these layers work by binning given random data set into cells or regions. These techniques avoid over plotting of individual samples and help visualize how data is distributed among the regions. But these visualization techniques are not suitable to understand spatial distribution of data. Depending on how boundaries are defined, visuals can change drastically between boundaries and hence they lack visualization of `continuity` and `pattern`. [heat map](https://en.wikipedia.org/wiki/Heat_map) provides solution by smoothening these boundaries. This RFC proposes a new `HeatmapLayer` to render heat maps.
+
+
+## Approach
+
+Grid aggregation is first used to aggregate data into small grid cells and generate a `Texture` object where each pixel represents a cell. Then using [KDE](https://en.wikipedia.org/wiki/Kernel_(statistics%29#Kernel_functions_in_common_use) function data is smoothed. Finally we render a textured rectangle covering the bounding box of input data samples using Texture produced above to produce the heat map. Following section describe these steps with more details.
+
+
+## Implementation
+
+### Possible Layer props
+
+* `data`
+
+* `longitudeRange` (Array, optional)
+  Range, [minLng, maxLng], of longitude values, any data sample outside of this range is excluded.
+
+* `lattiudeRange` (Array, optional)
```

Comment:
Agree, I can change these to `xRange`/`yRange` or `horizontalRange`/`verticalRange`. Thought of using `boundingBox` but it has to be an object with 4 keys, flat array might be confusing.

Reply:
What is the purpose of this prop? Should we automatically infer the bounding box from data/viewport?

Reply:
One of the data sample (used for Grid layer demo we have), I tried has few positions that are too far away from rest of them, these are either noise or invalid points. And this could happen for any other data sample and it leads to allocating textures of size more than 4K, hence app crashes. Also this is after normalizing all longitudes.